### PR TITLE
Toy Deploy

### PR DESCRIPTION
**Pull request by HeflerDev**
Deploying the toy project. I tried to do all the logic without any meaningful visuals, just to show all the practices that the project demanded.

**I am without a partner it's been almost 2 weeks since her Withdrawn, and I can't get a pairing, so I'm doing the projects alone.**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    @past_events = []
+    @upcoming_events = []
+    User.find(params[:id]).attended_events.each do |event|
+      if event.date < Time.now
+        @past_events << event.description
+      else
+        @upcoming_events << event.description
+      end
+    end
+  end
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: Can be replaced by 
```ruby
@user.past_events
@user.upcoming_events
```

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,6 +10,14 @@
   </head>
 
   <body>
+  <ul>
+    <li><%= link_to "New User", signup_path %></li>
+    <li><%= link_to "User Login", login_path %></li>
+    <% if logged_in? %>
+      <li><%= link_to "Create Event", host_path %></li>
+      <li><%= link_to "Index", events_home_path %></li>
+    <%end%>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb]: In that case, the session link will be available to the user.
Think about having something like this
```ruby
<% if logged_in? %>
  <li><%= link_to "Create Event", host_path %></li>
  <li><%= link_to "Index", events_home_path %></li>
<%else%>
<li><%= link_to "New User", signup_path %></li>
<li><%= link_to "User Login", login_path %></li>
<%end%>
```

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (Oluwadamilareolusakin) [reply app/controllers/users_controller.rb:28]: # Great work! @HeflerDev 
You did nicely implementing the conditional rendering for the login and logout buttons, however, the upcoming and past events scopes in the user's controller are still missing. Please note that even though you current solution works nicely, it won't be as performant as the solution suggested by @maelfosso. Kindly look into implementing this before your next review.


I wish you the very best and see you at the top! :rocket: 

##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -27,18 +27,18 @@
...
-            <%= event %>
+            <%= event.description %>
           </li>
         <%end%>
       <h2> Future Events </h2>
-        <% @upcoming_events.each do |event| %>
+        <% @user.created_events.upcoming.each do |event| %>
         <li>
-          <%= event %>
+          <%= event.description %>
```
Comment (samgaco) [inline app/views/users/show.html.erb]: :+1:

**Pull request merged**